test/framework/src/main/java/org/elasticsearch/ingest/TestProcessor.java

```diff
@@ -31,24 +32,30 @@
 
     private final String type;
     private final String tag;
-    private final Consumer<IngestDocument> ingestDocumentConsumer;
+    private final Function<IngestDocument, IngestDocument> ingestDocumentMapper;
```

Comment:
This requires adjusting `CompoundProcessorTests` which are currently failing tests.